src/lightning/pytorch/loops/training_epoch_loop.py

```
@@ -272,6 +275,23 @@ def advance(self, data_fetcher: _DataFetcher) -> None:
         # we are going to train first so the val loop does not need to restart
         self.val_loop.restarting = False
 
+        # =====================================================================
+        # FINAL: Check for SIGTERM broadcast and exit synchronously across ranks
+        from lightning.pytorch.utilities.exceptions import SIGTERMException
+
+        if dist.is_available() and dist.is_initialized() and self.trainer.world_size > 1:
+            with suppress(Exception):  # never crash CI
+                sigterm_tensor = torch.tensor(
+                    [1 if getattr(self.trainer, "received_sigterm", False) else 0],
+                    device=self.trainer.strategy.root_device,
+                )
+                dist.broadcast(sigterm_tensor, src=0)
+
+                if sigterm_tensor.item() == 1:
+                    dist.barrier()
+                    raise SIGTERMException()
```

How is this raised since all exceptions are ignored?